File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
// function solution(left, right) {
//     let answer=0;
//     for(let i=left; i<=right; i++){
//         let num=2;
//         for(let j=2; j<i; j++){
//             if(i%j===0) num++;
//         }
//         if(num%2===0) answer+=i;
//         else answer-=i;
//     }
//     return answer;
// }
function solution(left, right) {
    let answer = 0;

    for (let i = left; i <= right; i++) {
        // 제곱수인지 확인
        if (Number.isInteger(Math.sqrt(i))) {
            answer -= i; // 제곱수일 경우 약수의 개수가 홀수
        } else {
            answer += i; // 제곱수가 아닌 경우 약수의 개수가 짝수
        }
    }

    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer=0;
        for(int i=left; i<=right; i++){
            int cnt=get(i);
            if(cnt%2==0) answer+=i;
            else answer-=i;
        }
        return answer;
    }
    
    public static int get(int n){
        int total=1;
        for(int i=2; i<=n; i++){
            if(n%i==0) total++;
        }
        return total;
    }
}